### ISSUE TITLE
Implicit dependency on once_cell crate

### ISSUE THREAD

**Issue by farcaller** [yassun4dev/serde_valid#5]
It looks like this crate implicitly depends on `once_cell` being a dependency in

```rust
static #pattern_ident : ::once_cell::sync::OnceCell<::regex::Regex> = ::once_cell::sync::OnceCell::new();
```

But this isn't documented anywhere.

**Comment by yassun4dev** [comment]
Thank you for your suggestion.

The crate that serde_valid depends on is written in Cargo.toml, but there seems to be a culture in Rust to make it explicit in the documentation.

Do you know of any project or documentation rules that might be helpful? I would like to study them.

**Comment by farcaller** [comment]
I don't think it works this way. The macro unwraps not in the context of your crate, but the destination crate which might not have the once_cell dependency (mine didn't).

One option would be to re-export the symbols of once_cell through serde_vaild and then depend on them – because the serde_valid dependency is required to be explicit.

**Comment by yassun4dev** [comment]
@farcaller I released v1.12.0
